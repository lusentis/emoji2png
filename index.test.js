const Parser = require('.');

describe('Emoji parser', () => {
  it('should parse emoji', () => {
    const parser = new Parser();
    const result = parser.parse('Text ğŸ™‚ text');
    expect(result).toMatchSnapshot();
  });

  it('should not crash for null', () => {
    const parser = new Parser();
    const result = parser.parse(null);
    expect(result).toBe(null);
  });

  it('should not crash for undefined', () => {
    const parser = new Parser();
    const result = parser.parse(undefined);
    expect(result).toBe(undefined);
  });

  it('should not crash for empty string', () => {
    const parser = new Parser();
    const result = parser.parse('');
    expect(result).toBe('');
  });
});
