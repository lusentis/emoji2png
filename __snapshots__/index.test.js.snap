// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Emoji parser should parse emoji 1`] = `"Text <img draggable=\\"false\\" class=\\"emoji\\" src=\\"data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAACgAAAAoCAMAAAC7IEhfAAAC+lBMVEVHcEz5oADQdQvslQW+hD/AfTPJbAjMcAf/nQCxf0DLbwaxfD7ShCaqgU2xfEfOdArpkgb1kAWrekDdewa1eTbTkjfYgArumATLcAjVfxHAdSPSfA7KbgbbewnwmgfpkAbhiwe4eDC8ei3qhQrIeSHzkAjljBbzlQjzow3TkDffkSbffgXxmQbkmyjvphn3shDMlEj5ug3AcRzYfBPWdwflhxHqkhP3oAX2oAj7vg77vQ/XiiDYeQfsiQPrmg//uij/0Df/3jD+sRf/th7/rxj/tSH/0jb//LX/uCT/1zz/xDH/2DX0oxX/sxz/1jb/zS3wnxH/yTP/+qT/1DPTdwb/zzP/vjL/+IH9rRb/2zD/+qLeiAn/2TL/+HD/1jD//+ONTgL/yy//+ZThjAvPcgXypBP7qRbbgAb5qxTpmBD/0C//xCn//Mb7rhP//LD/1C7/+Yz/uwP/ySv4qBbxoRPlkQznlA//viv/vS7/y0L/7G//83L/5zL/92D//uD/8mf95Ur/7T7/8kjKpib/wS//xDb8qxbfgwL/+ZH/41X//cH/yTr/yzj/+6r/7353Qgn//br//cv/6Df/+Hf/+Xz/3j1+QgP/xi72qBTXdgHhjwz/4DH/xjT4qQ//wCnniwP/0ir/mgD/0EP/11P/+p7/5GL/+Yb//dH3mwfymQacbx3/7Fj//8//9lSRVQj///D/7E+eahL//lj//2//9T7aty3t2kinZQX/+pn6jwD/zzr/0z//xzn/3kv7mAP/3FL/22D/qgXZfAb/72XaqzbanSXiszv/2Cb/rwP/6TfBniz/2Ej/6mL/9YD1pQzfpCj/4F7tsy7tvjv//dSxcxWRZBz/53vvlAX/5nL1wz//8JT/9JD/twD/4zLGnDT/6jT/4kH42jHx0jH//mzt1TLpzTH//WGrfBaIRQD//EyFUgv/pgeZXA64ehj/tgd8TBDipyjPkCHLjR+AUBLTqTj/zhT/zRD/0BpeIgC4kB/jzEa4jx5fIQDhy0YO6+JtAAAAPXRSTlMAEM05MEbu5wEb/it/DQfHSO4k9zobhzLhko+e9OgiUmFYYeSY5cDi3FqL+yWXxeM88qLG88nD6Ob5+kj4kGx0ngAABFxJREFUeAGNlQN4W10AQOus1jrbxm9GjdcY5RDbTuattm3bnG3b5m8bN2/mt5Pk5eJc4zm9I8NeSfELcVu0yC3E7+3lPGfPX7Bw8aefLV64YP7skDdq3nNmaX9Fdeyy23d1oH7Rzprj9lrN1+cT7b5dUZnL2wHLW1JtkdqPfXxf9dxm7P/NlrF6+fLVj1i+OimtYv9Hr1TqPvNyblRLpkKRWV9/6lR9vSPUkrrl8kz3l7zAy4mqDBUen5GUpAAkJWXg8aoMfK028AXTe/rlbQxGTRQjNVUFbOCoUlMZUTVRqbXfTnd7bhwzdmy6eEmTwOMN1AAbEFVTM8DjJWguXaze8eGzEflUys+mpR0Jr6vTaBKA7lA0mrq68CNpaWfplT5PvKDA/dXnbTa7PTs7JycnHAIEsrPtdpvt/HZ0oPdjcUxl2fbjtYd3rVlzq7t76WO6u2+t2bXrcO3x7bGVox+v29SvrNXbEhObm1f1Hfzz7iqIu/8c7FvV3JyYuK3aumOqJyROHsRYqw5tWbu2Y23n7w/7uloBXX0P/24CCWu3HKqyYgenQOK0QezmoU3E3Ovre/Pa8g4+WA94cBAEe9dfzyVuGtpM2DoW2lfvb8WS6EjKktx1G1b0r/ijZx2g5y8Q3LAudwkFSScRtrr4AdFzfAO5LJ8rY8Yc6HXk9mQBekCZtt6smGUybr4Z2zA+CIhuwQ1oRJmOWqFG3bh3v/P2tSuAa7c779+7EamuoOrkCHTD3OFAHB7cqCSUyXVUmRp1c8OdlahiAGrlnQ03UZBXRlA2znWHxHMSrFmu48qAeeBKJAoi8soBlFotA6KZIGnc6RDdPjhXGmremM8F3Vy2LCYmJhIA/papmTIuN3+jObT03BeOpj1h5UfRiI0kOpfNZjKZSx4Dgmw2l07aiEAfLYcG4+dfTlNiLaQSOp2NxDGZFAgmE4dk0+klJAtWSSt38YUmXAwvRRM4sbF0NgWHxD0GiQRTGBvLIaBL4fHvPVrCQqNBGYrgxKac2IMk4oaQgCEkkXjsWEosB0FWCo2Fj5bQc1Q8vAANzM0nmzq/TyFCDJ04fWZ3MgeBRRfA42Gej7dZYRFN4jCTTzalt53+cc+eH775Ou/MXuCFoiUGY+GYJxvXQ2wSOkxLbNXPTT+lA747c9WabHF4QqnYw/vpURBES4USDJmAsCRXXdi7e/feC9YUC4JAxgAvWjDC6QnO/gK+VKiPwJCxwCWVlIDZI2DJ6Ai9EF7EcnGGJIiRMFaR1FAgAioZC0EmYyLiCgzSIpaHq9NzBHiw+FJamF4UEYGBiIgQ6cNoUj5r3MQX78kAGCvaCAeqXiSKixOJ9PowA9wUzYIB70Vc/b3EfJOURguDMNDgJr5Y4OLq9ArOI2ACMd9okkrhcKnUZASax4iQVy9lEB85epSXYKc4GiDeyfIaNXrkY+1VJkwa6w8b5+U1DuYydtIEqPgbcZ7nGhDgOs/5lVcBiLwT/wNQkeH9sDKk2gAAAABJRU5ErkJggg==\\"> text"`;
